src/main/java/duke/data/task/Deadline.java

```diff
@@ -0,0 +1,36 @@
+package duke.data.task;
+
+import duke.data.exception.InvalidDateInputException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
+
+    public Deadline(String description, String by) throws InvalidDateInputException {
+        super(description);
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new InvalidDateInputException(by);
+        }
+    }
+
+    public String getType() {
+        return "D";
+    }
+
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```

Comment:
Avoid having such complex nested logic in one line. Either separate it out into different lines or abstract it out into a different function